lib/components/map/default-map.tsx

```diff
@@ -319,6 +325,23 @@ class DefaultMap extends Component {
     const baseLayerUrls = baseLayersWithNames?.map((bl) => bl.url)
     const baseLayerNames = baseLayersWithNames?.map((bl) => bl.name)
 
+    const overlayTypes = overlays
+      ?.filter((overlay) => overlay?.type === 'otp2')?.[0]
+      ?.layers?.map((layer) => layer?.type)
+    const handleSetLocation = (location: MapLocationActionArg) => {
```

Comment:
This is one approach, but could we not fix this by adjusting query params? What if we check for map state in the places where query params are set? Might that be a better separation of concerns than setting modes within the map component?